Detector/DetStudies/src/components/FilterSiliconEcalHits.h

```diff
@@ -0,0 +1,70 @@
+#ifndef DETSTUDIES_FilterSiliconEcalHits_H
+#define DETSTUDIES_FilterSiliconEcalHits_H
+
+// GAUDI
+#include "GaudiAlg/GaudiAlgorithm.h"
+#include "GaudiKernel/ServiceHandle.h"
+
+// FCCSW
+#include "FWCore/DataHandle.h"
+#include "TH2F.h"
+#include "TTree.h"
+#include "TF1.h"
+#include "TProfile.h"
+
+class IGeoSvc;
+
+// datamodel
+namespace fcc {
+class PositionedCaloHitCollection;
+}
+
+class ITHistSvc;
+/** @class DECalAnalysis DECalAnalysis.h
+ *
+ *  Histograms of energy deposited in active material and total energy deposited in the calorimeter.
+ *  Passive material needs to be marked as sensitive. It needs to be divided into layers (cells) as active material.
+ *  Layers (cells) are numbered starting at 1 so that energy depoisted in cryostat and bath could be easily recognised.
+ *  Sampling fraction is calculated for each layer as the ratio of energy deposited in active material to energy
+ *  deposited in the layer (also in passive material).
+ *
+ *  @author Anna Zaborowska
+ */
+
+class FilterSiliconEcalHits : public GaudiAlgorithm {
+public:
+  explicit FilterSiliconEcalHits(const std::string&, ISvcLocator*);
+  virtual ~FilterSiliconEcalHits();
+  /**  Initialize.
+   *   @return status code
+   */
+  virtual StatusCode initialize() final;
+  /**  Fills the histograms.
+   *   @return status code
+   */
+  virtual StatusCode execute() final;
+  /**  Finalize.
+   *   @return status code
+   */
+  virtual StatusCode finalize() final;
+
+
+private:
+  /// Pointer to the interface of histogram service
+  ServiceHandle<ITHistSvc> m_histSvc;
+  /// Pointer to the geometry service
+  ServiceHandle<IGeoSvc> m_geoSvc;
+
+  /// The Pixel analysis
+  /// Handle for the energy deposits
+  DataHandle<fcc::PositionedCaloHitCollection> m_deposits{"rec/deposits", Gaudi::DataHandle::Reader, this};
```

Comment:
That is maybe not a comment for an immediate change, but rather towards the further developments.
Do you need it to work on `fcc::PositionedCaloHitCollection`? Or is it enough on `fcc::CaloHitCollection`?
If on both, I'd suggest to template. In a long run, the digi + reco is normally done on `fcc::CaloHitCollection` and positions can be found for the modules based on cellid, so using `fcc::PositionedCaloHitCollection` means data multiplication.